Pass timestep to solve() function

Psirus opened the pull request:
In addition to `t`, this will pass `dt` to the `solve()` function. The build failures are due to hypothesis being upgraded to 4.13, and now testing additional checkpoints and some such. With the previous version (in my case 4.5) the tests would run.

Psirus committed:
Pass timestep to solve() function

TTitscher commented on tests/test_timestepping.py:
```diff
@@ -27,8 +27,7 @@ def set(self, dt, value):
 
-    def __call__(self, t):
-        dt = t - self.t
+    def __call__(self, t, dt):
```
As `dt` is now given, we can get rid of a whole bunch of stuff in this class. `self.t` is pointless now, as well as `get` and `set`.

TTitscher reviewed (approved):
I'm fine with that. What do we do about the tests?

TTitscher committed:
Modify tests so they work. :hushed:

- fix st.floats(max=0.0) to -0.0
- adapt DeterministicSolve so that it does not fail for dt < 1.e-5.
  This solves an issue where a checkpoint requires dt to be smaller than
  dt_min and DeterministicSolve randomly fails.
